lib/rust/ensogl/core/src/display/style/theme.rs

```diff
@@ -42,24 +42,29 @@ impl Theme {
         Self { tree, on_mut }
     }
 
-    /// Insert a new style in the theme. Returns [`true`] if the operation was successful. It can
-    /// fail if provided with malformed value, for example with a string "rgba(foo)". In such a
-    /// case, the value will not be applied and the function will return [`false`].
-    pub fn set<P, E>(&self, path: P, value: E) -> bool
-    where
-        P: Into<Path>,
-        E: TryInto<Value>, {
-        let path = path.into();
-        let value = value.try_into();
-        if let Ok(value) = value {
-            self.tree.borrow_mut().set(&path.rev_segments, Some(value));
-            self.on_mut.run_all();
+    /// Insert or modify a style in the theme. Sets the style and returns [`true`] if `value` was
+    /// successfully parsed to [`Data`], or returns [`false`] otherwise.
+    pub fn set_parsed(&self, path: impl Into<Path>, value: &str) -> bool {
+        let parsed_value = value.parse::<Value>();
+        if let Ok(value) = parsed_value {
+            self.set(path, value);
```

Comment:
This name of the function suggests that we set something parsed. In reality, we are parsing it before setting.